### PR TITLE
Orchestrate versions

### PR DESCRIPTION
**Pull request by clebs**
Confirm these statements before you submit your pull request:

- [x] I have read and submitted the required [Contributor Licence Agreements](https://github.com/kyma-project/community/blob/master/CONTRIBUTING.md#agreements-and-licenses).
- [x] This pull request follows the contributing guidelines, recommended Git workflow, and templates.
- [x] I have tested my changes.
- [x] I have updated the relevant documentation.
---

**Description**

Changes proposed in this pull request:

- The orchestrator now tracks and passes on individual component versions to the pipelines.
- Docker images are now labeled with component and version metadata
- Latest docker images are now tagged after a successful master component build

### CODE REVIEW

File: docs/Jenkinsfile [hunk 1]
```diff
@@ -142,6 +142,22 @@ podTemplate(label: label) {
 
                             sh "docker tag $authorization_and_authentication-$docs_image_suffix ${dockerPushRoot}$authorization_and_authentication-$docs_image_suffix:$dockerImageTag"
                             sh "docker push ${dockerPushRoot}$authorization_and_authentication-$docs_image_suffix:$dockerImageTag"
+                            if (isMaster) {
+                                sh "docker push ${dockerPushRoot}$kyma-$docs_image_suffix:latest"
```
Comment (suleymanakbas91) [inline docs/Jenkinsfile]: These images should be built or tagged first with `latest` before pushing.

Comment (clebs) [reply docs/Jenkinsfile]: Yes definitely, missed that! Fixed now thanks!

File: components/helm-broker/Jenkinsfile [hunk 2]
```diff
@@ -62,22 +62,28 @@ podTemplate(label: label) {
                                 sh "cp ./indexbuilder deploy/tools/indexbuilder"
                                 sh "cp ./targz deploy/tools/targz"
 
-                                sh "docker build -t helm-broker:latest deploy/broker"
-                                sh "docker build -t helm-broker-reposerver:latest deploy/reposerver"
-                                sh "docker build -t helm-broker-tools:latest deploy/tools"
+                                sh "docker build -t ${dockerPushRoot}helm-broker:latest deploy/broker --label version=${dockerImageTag} --label component=${application}"
```
[the remark this answers is not in the document]
Comment (clebs) [reply components/helm-broker/Jenkinsfile]: This label is there to be able to group images per component in cases like helm broker, where several images are pushed.
It is not used yet, but will be helpful in the future.

File: Jenkinsfile [hunk 3]
```diff
@@ -77,7 +81,8 @@ if(isMaster && currentBuild.currentResult == "SUCCESS") {
     stage("trigger remote cluster"){
         build job: 'azure/master', parameters: [
             string(name:'REVISION', value: "${commit_id}"),
-            string(name:'APP_VERSION', value: "${params.APP_VERSION}")],
+            string(name:'APP_VERSION', value: "${params.APP_VERSION}"),
+            string(name:'COMP_VERSIONS', value: "${params.COMP_VERSIONS}")],
```
[the remark this answers is not in the document]
Comment (clebs) [reply Jenkinsfile]: I am just passing the parameter for now, complete usage of these versions will be done here:
https://jira.hybris.com/browse/YSF-2892

[the remark this answers is not in the document]
Comment (aszecowka) [reply Jenkinsfile]: thanks for clarification, I am not sure if we are allowed to point to jira here ;)

**Pull request merged**